runtime/src/bank/pyth_accumulator.rs

```diff
@@ -0,0 +1,369 @@
+use {
+    super::Bank,
+    crate::accounts_index::{ScanConfig, ScanError},
+    byteorder::LittleEndian,
+    byteorder::ReadBytesExt,
+    log::*,
+    pyth_oracle::validator::AggregationError,
+    pythnet_sdk::{
+        accumulators::{merkle::MerkleAccumulator, Accumulator},
+        hashers::keccak256_160::Keccak160,
+        wormhole::{AccumulatorSequenceTracker, MessageData, PostedMessageUnreliableData},
+    },
+    solana_sdk::{
+        account::{AccountSharedData, ReadableAccount},
+        borsh, feature_set,
+        hash::hashv,
+        pubkey::Pubkey,
+    },
+    std::env::{self, VarError},
+};
+
+pub const ACCUMULATOR_RING_SIZE: u32 = 10_000;
+
+lazy_static! {
+    pub static ref MESSAGE_BUFFER_PID: Pubkey = env_pubkey_or(
+        "MESSAGE_BUFFER_PID",
+        Pubkey::new_from_array(pythnet_sdk::MESSAGE_BUFFER_PID),
+    );
+    pub static ref ACCUMULATOR_EMITTER_ADDR: Pubkey = env_pubkey_or(
+        "ACCUMULATOR_EMITTER_ADDR",
+        Pubkey::new_from_array(pythnet_sdk::ACCUMULATOR_EMITTER_ADDRESS),
+    );
+    pub static ref ACCUMULATOR_SEQUENCE_ADDR: Pubkey = env_pubkey_or(
+        "ACCUMULATOR_SEQUENCE_ADDR",
+        Pubkey::new_from_array(pythnet_sdk::pythnet::ACCUMULATOR_SEQUENCE_ADDR),
+    );
+    pub static ref WORMHOLE_PID: Pubkey = env_pubkey_or(
+        "WORMHOLE_PID",
+        Pubkey::new_from_array(pythnet_sdk::pythnet::WORMHOLE_PID),
+    );
+    pub static ref ORACLE_PID: Pubkey = env_pubkey_or(
+        "ORACLE_PID",
+        "FsJ3A3u2vn5cTVofAjvy6y5kwABJAqYWpe4975bi2epH"
```

Comment:
not for now, but seeing this i'm wondering whether in the future it makes sense to move the entire code in this file to pythnet_sdk